### DISCUSSION TITLE
Forwarding ports of other docker-compose services

### DISCUSSION THREAD

**Discussion by jpwilliams** [community/community#26051] [Codespaces]
<p>I’m attempting to set up a development environment for a Node.js library that interfaces with RabbitMQ, the configuration for which is in the spoiler below.</p>
<details>
<summary>
Configuration files</summary>
<p>I have a simple Dockerfile:</p>
<pre><code class="lang-auto">ARG VARIANT=14
FROM mcr.microsoft.com/vscode/devcontainers/javascript-node:0-${VARIANT}
</code></pre>
<p>A <code>docker-compose.yml</code> file that boots up two services: <code>main</code> and <code>rabbitmq</code>:</p>
<pre><code class="lang-auto">version: "3"
services:
  main:
    build:
      context: .
      dockerfile: Dockerfile
    volumes:
      - ..:/workspace:cached
    command: sleep infinity
    ports:
      - 3000

  rabbit:
    image: rabbitmq:3-management
    restart: unless-stopped
    ports:
      - 5672
      - 15672

</code></pre>
<p>And finally a <code>devcontainer.json</code> file:</p>
<pre><code class="lang-auto">{
  "name": "Node.js &amp; RabbitMQ",
  "dockerComposeFile": "docker-compose.yml",
  "service": "main",
  "workspaceFolder": "/workspace",
  "settings": {
    "terminal.integrated.shell.linux": "/bin/bash"
  }
}
</code></pre>
</details>
<p>This works wonderfully and the <code>main</code> service that I use to develop on can successfully reach the <code>rabbit</code> service and interact with it.</p>
<p><strong>However, the RabbitMQ server exposes a helpful UI at port 15672 which I would like to be able to see, but I don’t know if it’s possible for me to get at it.</strong></p>
<p>I can add <code>"forwardPorts": [15672]</code> to my <code>devcontainer.json</code>, but this seems to forward that port to my <code>main</code> service, when I want it to go to the <code>rabbit</code> service.</p>
<p>Is there any way to achieve this?</p>
<p>P.S. I’ve been eagerly awaiting Codespaces. I’m a huge fan of GitPod and this solution seems absolutely wonderful! Fantastic work; I can’t wait to get using it properly and start submitting feedback! :heart:</p>

**Comment by Chuxel** [comment, marked as answer]
<p>You can resolve this with a network trick - we’re going to update the samples at vscode-dev-containers with this.</p>
<p>Try adding the following to your main service:</p>
<pre><code class="lang-yaml">network_mode: service:rabbit
</code></pre>
<p>That should make rabbit MQ available on localhost so the forwarding works properly.</p>

**Comment by jpwilliams** [comment]
<p>Fantastic! Works a charm. Thanks for such a speedy reply, <a class="mention" href="/chuxel">@Chuxel</a>, and thank you for doing amazing work with Codespaces!</p>
